File: 3676-smallest-number-with-all-set-bits/smallest-number-with-all-set-bits.py
class Solution:
    def smallestNumber(self, n: int) -> int:
        l, r = 1, 10
        res = -1
        while l <= r:
            m = (l + r) // 2
            x = 2 ** m - 1
            if x >= n:
                r = m - 1
                res = x
            else:
                l = m + 1
        return res

File: 3676-smallest-number-with-all-set-bits/smallest-number-with-all-set-bits.java
class Solution {     
    boolean isAllBitsSet(int x) {         
        return (x & (x + 1)) == 0;     
    }     

    public int smallestNumber(int n) {         
        int result = n;         
        while (!isAllBitsSet(result)) {             
            result++;         
        }         
        return result;     
    } 
}
